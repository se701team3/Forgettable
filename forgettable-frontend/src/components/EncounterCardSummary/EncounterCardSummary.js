import React from 'react';
import {Avatar} from '@mui/material';
import classes from './EncounterCardSummary.module.css';
import {getLongDateStringWithSlashes} from '../../functions/dateFormatter';
import UnknownDetail from '../UnknownDetail/UnknownDetail';
import PropTypes from 'prop-types';
import {getImageSrcFromBuffer} from '../../functions/getImageSrcFromBuffer';

/*
 * Component for displaying information of an encounter in a list.
 * This is the small version of the encounter card.
 *
 * Author: Mercury Lin (lin8231)
 */
const EncounterCardSummary = (props) => {
  return (
    <div
      className={classes.EncounterCardSummary}
      onClick={props.onClick}
      data-testid="container-card"
    >
      <div className={classes.ContentContainer}>
        <div className={classes.HeaderContainer}>
          <Avatar
            alt={props.name}
            src={getImageSrcFromBuffer(props.img)}
            sx={{
              height: '70px',
              width: '70px',
              marginRight: '14px',
              backgroundColor:
                  getComputedStyle(document.body).getPropertyValue('--prmry'),
              fontSize:
                  getComputedStyle(document.body)
                      .getPropertyValue('--text-xxlarge'),
            }}
          />
          <div className={classes.IdentityInfoConatiner}>
            <h3 data-testid="name-element">{props.name}</h3>
            <p data-testid="location-element">
              Met at: {props.location || 'some cool place'}
            </p>
          </div>
        </div>
        <div className={classes.DetailsContainer}>
          <p >
            {'Date you met: '}
            {props.dateMet ?
           <span data-testid="date-met-element">
             { getLongDateStringWithSlashes(props.dateMet)}
           </span> :
           <UnknownDetail/>}
          </p>
          <p
            className={classes.DescriptionText}
            data-testid="description-element">
            {props.description}
          </p>
        </div>
      </div>
    </div>
  );
};

EncounterCardSummary.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  img: PropTypes.string,
  dateMet: PropTypes.instanceOf(Date),
  description: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  location: PropTypes.string,
};

export default EncounterCardSummary;
