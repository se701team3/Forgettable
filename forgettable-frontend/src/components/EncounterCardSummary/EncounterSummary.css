@import url('https://fonts.googleapis.com/css2?family=Poppins&display=swap');
@import  '../../colours.css';

 .container{
    display:flex;
    flex-direction: column;
    align-items: center;
}
.card{
    border-radius: 13;
    height: 220px;
    width:245px;
}
.header{
    display: flex;
    flex-direction: row;
    padding-right:20px;
    padding-left:20px;
    padding-top: 20px;
    font-weight:500;
    font-size: 20px;
    font-family:'Poppins', sans-serif;
    line-height: 30px;
    color: var(--ltxt1);
}
.description{
    font-family: 'Poppins', sans-serif;
    font-size: 13px;
    font-weight: 400;
    line-height: 21px;
    color: var(--litext2);
    font-style: normal;
}
.nameDesc{
    padding-left: 8px;
}
.date{
    display: flex;
    flex-direction: row;
    padding-left: 20px;
    padding-right:20px;
    padding-top:10px;
    font-size: 13px;
    font-family: 'Poppins', sans-serif;
    font-weight: 400;
    color:var(--litext2);
}
.descriptionText{
    padding-left: 20px;
    padding-right: 20px;
    padding-top: 10px;
    margin-bottom:10px;
    font-size: 13px;
    line-height: 21px;
    font-family: Poppins;
    font-weight: 400;
    color:var(--litext2);
}
