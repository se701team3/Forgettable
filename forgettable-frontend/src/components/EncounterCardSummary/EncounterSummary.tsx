import * as React from 'react';
import Card from '@mui/material/Card';
import {Avatar} from '@mui/material';
import classes from './EncounterSummary.module.css';

export default function EncounterSummary(props: { Name: any; Date: any; Description: any; Summary: any; Src: string}) {
  const {
    Name, Date, Description, Summary, Src,
  } = props;

  return (
    <div className={classes.Container}>
      <Card
        className={classes.Card}
        sx={{maxWidth: 220, maxHeight: 240, bgcolor: 'var(--lcard)', borderRadius: 6, boxShadow: 0}}
      >
        <div className={classes.Header}>
          <Avatar
            alt={Name}
            src={Src}
            sx={{width: 70, height: 70}}
          />
          <div className={classes.NameDesc}>
            <div>{Name}</div>
            <div className={classes.Description}>{Description}</div>
          </div>
        </div>
        <div className={classes.Date}>
          <div>Date you met:&nbsp;</div>
          {Date}
        </div>
        <div className={classes.DescriptionText}>
          {Summary}
        </div>
      </Card>
    </div>
  );
}
