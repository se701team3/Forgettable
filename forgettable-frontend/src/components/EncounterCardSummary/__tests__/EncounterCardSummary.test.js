import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import {render, fireEvent, cleanup} from '@testing-library/react';
import {
  within,
  getByText,
  getByTestId,
} from '@testing-library/dom';
import EncounterCardSummary from '../EncounterCardSummary';

const FULL_NAME = 'Mercury Element';

it('renders EncounterCardSummary UI with correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<EncounterCardSummary
    name={FULL_NAME}
    description="I met her at a mall"
    onClick={jest.fn()}
  />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});

afterEach(cleanup);

it('Name prop passed in is displayed correctly', () => {
  const {getByTestId} = render(
      <EncounterCardSummary
        name={FULL_NAME}
        description="I met her at a mall"
        onClick={jest.fn()}
      />);

  const {getByText} = within(getByTestId('name-element'));

  expect(getByText(FULL_NAME)).toBeInTheDocument();
});

it('displays the correct format for firstMet for an existing location', () => {
  const location = 'Auckland';

  const {getByTestId} = render(
      <EncounterCardSummary
        name={FULL_NAME}
        description="I met her at a mall"
        onClick={jest.fn()}
        location={location}
      />);

  const {getByText} = within(getByTestId('location-element'));

  expect(
      getByText('Met at: ' + location),
  ).toBeInTheDocument();
});

it('displays the date met in the correct format', () => {
  const date = new Date('01-01-2000');

  const {getByTestId} = render(
      <EncounterCardSummary
        name={FULL_NAME}
        description="I met her at a mall"
        onClick={jest.fn()}
        dateMet={date}
      />);

  const {getByText} = within(getByTestId('date-met-element'));

  expect(
      getByText('01/01/2000'),
  ).toBeInTheDocument();
});

it('displays the description when passed in as a prop', () => {
  const description = 'I met her at a mall';

  const {getByTestId} = render(
      <EncounterCardSummary
        name={FULL_NAME}
        description="I met her at a mall"
        onClick={jest.fn()}
      />);

  const {getByText} = within(getByTestId('description-element'));

  expect(
      getByText(description),
  ).toBeInTheDocument();
});


it('successfully fires event when card is clicked', () => {
  const handleClick = jest.fn();

  const {getByTestId} = render(
      <EncounterCardSummary
        name={FULL_NAME}
        description="I met her at a mall"
        onClick={handleClick}
      />);

  const node = getByTestId('container-card');

  fireEvent.click(node);
  expect(handleClick).toHaveBeenCalledTimes(1);
});


