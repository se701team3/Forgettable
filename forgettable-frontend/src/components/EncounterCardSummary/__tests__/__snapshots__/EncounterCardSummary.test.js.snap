// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EncounterCardSummary UI with correct hierarchy 1`] = `
<div
  className="EncounterCardSummary"
  data-testid="container-card"
  onClick={[MockFunction]}
>
  <div
    className="ContentContainer"
  >
    <div
      className="HeaderContainer"
    >
      <ForwardRef(Avatar)
        alt="Mercury Element"
        src={null}
        sx={
          Object {
            "backgroundColor": "",
            "fontSize": "",
            "height": "70px",
            "marginRight": "14px",
            "width": "70px",
          }
        }
      />
      <div
        className="IdentityInfoConatiner"
      >
        <h3
          data-testid="name-element"
        >
          Mercury Element
        </h3>
        <p
          data-testid="location-element"
        >
          Met at: 
          some cool place
        </p>
      </div>
    </div>
    <div
      className="DetailsContainer"
    >
      <p>
        Date you met: 
        <UnknownDetail />
      </p>
      <p
        className="DescriptionText"
        data-testid="description-element"
      >
        I met her at a mall
      </p>
    </div>
  </div>
</div>
`;
